src/main/scala/zio/s3/Live.scala

```diff
@@ -105,78 +108,78 @@ final class Live(unsafeClient: S3AsyncClient) extends S3.Service {
       .flatMap(publisher =>
         execute(
           _.putObject(
-            PutObjectRequest
-              .builder()
+            UploadOptions
+              .putObjectBuilder(options)
               .bucket(bucketName)
               .contentLength(contentLength)
-              .contentType(contentType)
               .key(key)
-              .metadata(metadata.asJava)
               .build(),
             AsyncRequestBody.fromPublisher(publisher)
           )
         )
       )
       .unit
 
-  // only for file which are bigger than 5 Mb
   def multipartUpload[R <: zio.Has[_]: Tag](
     bucketName: String,
     key: String,
-    contentType: String,
     content: ZStream[R, Throwable, Byte],
-    metadata: Map[String, String] = Map.empty
+    options: UploadOptions = UploadOptions.default,
+    partSize: Int = UploadOptions.MinMultipartPartSize,
+    parallelism: Int = 1
   ): ZIO[R, S3Exception, Unit] =
```

Comment:
i would be more in favor to keep this definition

```scala
 def multipartUpload[R <: zio.Has[_]: Tag](
    bucketName: String,
    key: String,
    content: ZStream[R, Throwable, Byte],
    options: UploadOptions = MultiPartUploadOptions.default)(parallelism: Int = 1) 
```

Reply:
why is parallelism different compared to the other options? And where should the "partSize" be placed in this case?

Reply:
i prefer to separate to make it more obvious this is a special parameter. parallelism can have a big impact on performance/behavior of your application same as

```scala
mapMPar(n: Int)(f: O => ZIO[R1, E1, O2])
```

Reply:
I guess that's because the last arguments can become a code block
```
mapMPar(4) { value =>
   ....
}
```

Reply:
yep mapMPar have a better syntax :)

```scala
val content: ZStream[Any, Nothing, String] = ???

// more obvious 10 is a special value
multipartUpload("bucket1", "key1", s)(10)

//les obvious
multipartUpload("bucket1", "key1", s, 10)
```

Reply:
while adding the multipart specific options it seemed to me that putting parallelism there looks nice. So the parallelism is now considered a specific option as well. See https://github.com/zio/zio-s3/pull/117/commits/7f6632881537df013d77038a4e2f486a2a1dbb7a